### DISCUSSION TITLE
Search Dialog

### DISCUSSION THREAD

**Discussion by r11d11** [devemux86/cruiser#341] [Q&A] [answered]
Cruiser 3.0.33-b4

As a motorcyclist, I often look for a rest area with outdoor catering.
In the cruiser search dialog, I have the standard entries for FOOD, COFFEE and DRINK.
Would it be possible to add a search entry for beer garden/outdoor gastronomy?

I have started two queries in https://overpass-turbo.eu/
Maybe this helps to understand what I want:

1.)
/*
This has been generated by the overpass-turbo wizard.
The original search was:
""outdoor catering""
*/
[out:json][timeout:25];
// gather results
(
  // query part for: ""outdoor_seating""
  node["leisure"="outdoor_seating"]({{bbox}});
  way["leisure"="outdoor_seating"]({{bbox}});
  relation["leisure"="outdoor_seating"]({{bbox}});
);
// print results
out body;
>;
out skel qt;

and 2.) Biergarten or beer garden

/*
This has been generated by the overpass-turbo wizard.
The original search was:
"beer garden"
*/
[out:json][timeout:25];
// gather results
(
  // query part for: "beer garden
  node["amenity"="beer garden"]({{bbox}});
  way["amenity"="biergarten"]({{bbox}});
  relation["amenity"="beer garden"]({{bbox}});
);
// print results
out body;
>;
out skel qt;

**Comment by devemux86** [comment]
@r11d11 thanks for the suggestion.

You probably mean the offline search?
POI currently have all the data mentioned [here](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml). We can always improve it and add more data.

I see the `amenity=beer_garden` already exists. Is that the correct tag or [amenity=biergarten](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbiergarten)?
We could add it in the `Food` category?

POI categories in Cruiser offline search currently use the tags mentioned here:
- https://github.com/devemux86/cruiser/discussions/195#discussioncomment-6451058

---

Cruiser is extremely versatile, we can search for everything even without POI categories.
For example:

<img src="https://github.com/devemux86/cruiser/assets/3484020/1d21345b-327d-467b-b54b-93db54acaee2" width="50%"/>

**Comment by r11d11** [comment]
Yes I am talking about the offline search.
I know that I can seach for Biergarten but that does not always give the desired result

Example

I typed in Biergarten in Cruiser and the red dots in picture are the found Biergarten in Cruiser. I think they have the Biergarten in it's name

![grafik](https://github.com/devemux86/cruiser/assets/101214119/f5dd44ad-a054-4e47-b3d3-e1910d5ccd0e)

But where you can see the red arrow, there is the Hellers Volskarten (PlusCode WWCW+HH Köln)
Hellers Volksgarten is a nice Biergarten

And Hellers Volksgarten is tagged with amenity Biergarten in OSM. 
https://www.openstreetmap.org/?mlat=50.921373471464506&mlon=6.946524381637574#map=19/50.92145/6.94652

But i don't know how to find Hellers Volksgarten with Cruiser when I dont know the name

I thougt when there is a new category beer garden/outdoor gastronomy it would be easy
And I  would prefer not too mix is with food because food gives hundrets of result and it is difficult to find the one Biergarten in the search

But it would be a first step to add [amenity=biergarten](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbiergarten) to the food search. 
That would help me

**Reply by devemux86** [reply]
[I added](https://github.com/mapsforge/mapsforge/commit/c011e688ed931ea3e32be3ff3114a9edc91b6b5e) the amenity=biergarten in the poi-mapping and it will be included in the next update of the POI downloads.

For now, where should we put it, in Food or Drink?

---

About the POI categories, if we have a category for each tag the list will become very long.
We need to group somehow the related ones, e.g. shops types number is huge.

If the tag is included, we can search directly for the tag "biergarten".

Or we could use another UI where click on categories shows the sub-categories and then search.

**Comment by r11d11** [comment]
Ok, i understand that we need some grouping. Then I would prefer to add the tag "biergarten" to Food.
Categories with sub-categories is a good idea 👍

**Reply by devemux86** [reply]
It is already added in "Food" in the latest:
- https://github.com/devemux86/cruiser/discussions/329

Offline search will find it with the next poi updates.

**Comment by devemux86** [comment, marked as answer]
> Then I would prefer to add the tag "biergarten" to Food.

Available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
